### ISSUE TITLE
Fix escaping of aggregate output field names.

### ISSUE THREAD

**Issue by iaindillingham** [vega/vega#1810]
The pivot transform fails when there are apostrophes in field values. It fails silently, however, meaning that whilst the console reports an error (an unexpected identifier) the data viewer reports that the table is empty.

If you paste the following into the Vega Editor and open the data viewer, then the pivoted table will be empty. If you remove the apostrophe from Côte d'Ivoire, then the pivoted table will have a single row with three columns.

```json
{
  "$schema": "https://vega.github.io/schema/vega/v5.json",
  "data": [
    {
      "name": "table",
      "values": [
        {
          "country": "Liberia",
          "region": "West Africa",
          "number_of_apostrophes": 0
        },
        {
          "country": "Côte d'Ivoire",
          "region": "West Africa",
          "number_of_apostrophes": 1
        }
      ]
    },
    {
      "name": "pivoted",
      "source": "table",
      "transform": [
        {
          "type": "pivot",
          "field": "country",
          "value": "number_of_apostrophes",
          "groupby": [
            "region"
          ]
        }
      ]
    }
  ]
}
```

**Comment by jheer** [comment]
Thanks for the issue! I fear that this may not actually be a problem with the pivot transform, but potentially a lower-level issue with field name path parsing. Either way, I will investigate further when I can!

**Comment by jheer** [comment]
Thanks again for the issue. "Fortunately" the bug turns out to be a simple oversight in the aggregate code generation logic. Will be fixed by #1952.
